src/components/carousel/nz-carousel.component.ts

```diff
@@ -83,14 +96,18 @@ export class NzCarouselComponent implements AfterViewInit, OnDestroy {
       if (this.nzAutoPlay) {
         this.createInterval();
       }
-      this._renderer.setStyle(this.slickList.nativeElement, 'height', `${this.hostElement.nativeElement.offsetHeight}px`);
+
       if (this.nzVertical) {
+        this._renderer.setStyle(this.slickList.nativeElement, 'height', `${this.hostElement.nativeElement.offsetHeight}px`);
```

Comment:
slickList's height is defined when ngAfterViewInit trigger, this height won't change when vertical mode. If remove it style here, the height will be multiple here.